### PR TITLE
feat: add github action to build docker for release 

### PR DESCRIPTION
**Pull request by phv2312**
## Description

- Add github actions to automatically build and push docker to dockerhub for release
- To run, you need to set up your repository secret DOCKERHUB_USERNAME and DOCKERHUB_TOKEN in Settings

## Type of change

- [x] New features (non-breaking change).
- [ ] Bug fix (non-breaking change).
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected).

## Checklist

- [x] I have performed a self-review of my code.
- [ ] I have added thorough tests if it is a core feature.
- [ ] There is a reference to the original bug report and related work.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] The feature is well documented.

### CODE REVIEW

**Comment by ducminhle** [comment]
- Why don't we push Container images to Github Container Registry like https://github.com/runatlantis/atlantis/pkgs/container/atlantis?

File: .github/workflows/build-push-docker.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        id: buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Setup tag version
+        id: generate_tag
+        shell: bash
+        run: |
+          if [ "${{ github.ref_type }}" == "tag" ]; then
+            export IMAGE_TAG="${{ github.ref_name }}-$(echo $GITHUB_SHA | cut -c 1-7)"
+            echo "release=true" >> $GITHUB_OUTPUT
+          elif grep -q "main" <<< "${{ github.ref }}"; then
```
Comment (ducminhle) [inline .github/workflows/build-push-docker.yaml]: I believe this method of tagging container images is not suitable for open-source projects.

**Comment by phv2312** [comment]
> * Why don't we push Container images to Github Container Registry like https://github.com/runatlantis/atlantis/pkgs/container/atlantis?

I'm quite not familar with github container registry. Don't know Cinnamon has already have a place on it or should we create freshly ? And the current manual docker has already been pushed on Dockerhub already? 

But in depends on the PIC, I think switching between Dockerhub or Github registry does not take much effort

File: .github/workflows/build-push-docker.yaml (hunk 1 above)
Comment (phv2312) [reply .github/workflows/build-push-docker.yaml]: Can you suggest one @ducminhle ? Should we use the tag name only ( without sha ) ?

Comment (taprosoft) [reply .github/workflows/build-push-docker.yaml]: Glad to hear your suggestion on this @ducminhle

Comment (ducminhle) [reply .github/workflows/build-push-docker.yaml]: 1. we should tag container image with tag (without commit hash, ex: `v1.0.0`) and latest for the last image.
2. we don't need `pr-<commit hash>`.
3. main => `main-<commit hash>`.

**Comment by ducminhle** [comment]
`GitHub Container Registry is free for public images`, @phv2312 
https://docs.github.com/en/billing/managing-billing-for-github-packages/about-billing-for-github-packages
https://github.blog/news-insights/product-news/introducing-github-container-registry/

**Review by taprosoft — APPROVED** [review]
LGTM tks @phv2312

**Pull request merged**